client/client_test.go

```diff
@@ -1744,6 +1746,135 @@ func readFileInImage(c *Client, ref, path string) ([]byte, error) {
 	return ioutil.ReadFile(filepath.Join(destDir, filepath.Clean(path)))
 }
 
+func testMultiCacheImportExport(t *testing.T, sb integration.Sandbox) {
+	requiresLinux(t)
+	registry, err := sb.NewRegistry()
+	if errors.Cause(err) == integration.ErrorRequirements {
+		t.Skip(err.Error())
+	}
+	require.NoError(t, err)
+
+	c, err := New(context.TODO(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	busybox := llb.Image("busybox:latest")
+	st := llb.Scratch()
+
+	run := func(cmd string) {
+		st = busybox.Run(llb.Shlex(cmd), llb.Dir("/wd")).AddMount("/wd", st)
+	}
+
+	run(`sh -c "echo -n foobar > const"`)
+	run(`sh -c "cat /dev/urandom | head -c 100 | sha256sum > unique"`)
+
+	def, err := st.Marshal()
+	require.NoError(t, err)
+
+	target := registry + "/buildkit/testmultiexportinline:latest"
+
+	localCacheDir, err := ioutil.TempDir("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(localCacheDir)
+	resp, err := c.Solve(context.TODO(), def, SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type: ExporterImage,
+				Attrs: map[string]string{
+					"name": target,
+					"push": "true",
+				},
+			},
+		},
+		CacheExports: []CacheOptionsEntry{
+			{
+				Type: "inline",
+			},
+			{
+				Type: "local",
+				Attrs: map[string]string{
+					"dest": localCacheDir,
+				},
+			},
+		},
+	}, nil)
+	require.NoError(t, err)
+
+	dgst, ok := resp.ExporterResponse["containerimage.digest"]
+	require.Equal(t, ok, true)
+
+	unique, err := readFileInImage(c, "dummy@"+dgst, "/unique")
+	require.NoError(t, err)
+
+	err = c.Prune(context.TODO(), nil, PruneAll)
+	require.NoError(t, err)
+
+	checkAllRemoved(t, c, sb)
+
+	resp, err = c.Solve(context.TODO(), def, SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type: ExporterImage,
+			},
+		},
+		CacheExports: []CacheOptionsEntry{
+			{
+				Type: "inline",
+			},
+		},
+		CacheImports: []CacheOptionsEntry{
+			{
+				Type: "registry",
+				Attrs: map[string]string{
+					"ref": target,
+				},
+			},
+		},
+	}, nil)
+	require.NoError(t, err)
+
+	dgst2, ok := resp.ExporterResponse["containerimage.digest"]
+	require.Equal(t, ok, true)
+
+	require.Equal(t, dgst, dgst2)
+
+	err = c.Prune(context.TODO(), nil, PruneAll)
+	require.NoError(t, err)
+
+	checkAllRemoved(t, c, sb)
+
+	resp, err = c.Solve(context.TODO(), def, SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type: ExporterImage,
+			},
+		},
+		CacheExports: []CacheOptionsEntry{
+			{
+				Type: "inline",
+			},
+		},
+		CacheImports: []CacheOptionsEntry{
+			{
+				Type: "local",
+				Attrs: map[string]string{
+					"src": localCacheDir,
+				},
+			},
+		},
+	}, nil)
+	require.NoError(t, err)
+
+	dgst3, ok := resp.ExporterResponse["containerimage.digest"]
+	require.Equal(t, ok, true)
+
+	unique3, err := readFileInImage(c, "dummy@"+dgst3, "/unique")
+	require.NoError(t, err)
+
+	// dgst3 != dgst, because inline cache is not exported for local
+	require.Equal(t, unique, unique3)
```

Comment:
@tonistiigi Please confirm this test is correct?

Reply:
Why not export both inline and local?